### DISCUSSION TITLE
Infinite gass fee

### DISCUSSION THREAD

**Discussion by Taofeeq001** [Cyfrin/foundry-full-course-cu#2654] [General]
What might be the cause of getting infinite gas fee which cause my contract not to deploy?

**Comment by EngrPips** [comment]
What chain are you trying to deploy to? Some logic might be incorrect in your contract.

**Reply by Taofeeq001** [reply]
I am coding along because i have just started learning web3 using sol. 

function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
    the above function was the one showing infinite gas and i couldnt deploy it

**Reply by EngrPips** [reply]
Share your whole contract let me try deploy in remix.

**Reply by Taofeeq001** [reply]
// SPDX-License-Identifier:MIT

pragma solidity ^0.8.18;

contract SimpleStorage {
    uint256 myFavoriteNumber;

    struct Person {
        uint256 favoriteNumber;
        string name;
    }
    // uint256[] public anArray;
    Person[] public listOfPeople;

    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public virtual {
        myFavoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return myFavoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}

**Reply by EngrPips** [reply]
Deploys fine for me.
<img width="1440" alt="Screenshot 2024-10-10 at 09 39 33" src="https://github.com/user-attachments/assets/6694b1a5-83f5-4161-82bd-1c8cb6dd90aa">

**Reply by Taofeeq001** [reply]
thanks brother. i have been able to deploy it on zksinc now. However, does that mean i should not worry about getting infinite gas while writing contract?

**Reply by EngrPips** [reply]
Interesting, Keep crushing it!

**Reply by Taofeeq001** [reply]
Is there any community i can join for this platform ?

**Reply by EngrPips** [reply]
Yeah sure, here https://discord.gg/5q7AmaUq

**Reply by Taofeeq001** [reply]
Thanks a lot EngrPips

**Reply by EngrPips** [reply]
Anytime, Friend.
